### PR TITLE
Adds identify protocol

### PR DESCRIPTION
**Pull request by ralexstokes**
Adds the handler and supporting protobufs.

Supersedes #297.

Subsequent PRs will integrate `identify` into the `BasicHost` and also set it as a default protocol.

### CODE REVIEW

**Comment by ralexstokes** [comment]
note: just pulling this chunk out of #297 .

**Comment by mhchia** [comment]
Is it ready for review?

**Comment by ralexstokes** [comment]
@mhchia yep!

File: libp2p/identity/identify/protocol.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+def _multiaddr_to_bytes(maddr: Multiaddr) -> bytes:
+    return maddr.to_bytes()
+
+
+def identify_handler_for(
+    public_key: PublicKey, laddrs: Sequence[Multiaddr], protocols: Sequence[TProtocol]
+) -> StreamHandlerFn:
+    async def handle_identify(stream: INetStream) -> None:
+        peer_id = stream.muxed_conn.peer_id
+        logger.debug("received a request for % from %", ID, peer_id)
```
Comment (NIC619) [inline libp2p/identity/identify/protocol.py]: I got this error: `ValueError: unsupported format character '%'`

Comment (pipermerriam) [reply libp2p/identity/identify/protocol.py]: Those need to be `%s`, not just plain `%`

Comment (ralexstokes) [reply libp2p/identity/identify/protocol.py]: good catch!

**Comment by ralexstokes** [comment]
going to merge this in, there is a failure but it is the linter and being handled in #343.

should be able to wrap up this line of work now w/ #337 and the default protocols `ping` and `identify`.

**Pull request merged**